From: Nathan Walp <faceprint@faceprint.com>
Date: Thu, 1 Nov 2012 12:08:47 +0000
Subject: r8169: allow multicast packets on sub-8168f chipset.
Patch-mainline: v3.7-rc5
Git-commit: 0481776b7a70f09acf7d9d97c288c3a8403fbfe4
References: bnc#805371

RTL_GIGA_MAC_VER_35 includes no multicast hardware filter.

Signed-off-by: Nathan Walp <faceprint@faceprint.com>
Suggested-by: Hayes Wang <hayeswang@realtek.com>
Acked-by: Francois Romieu <romieu@fr.zoreil.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>

Conflicts:
	drivers/net/ethernet/realtek/r8169.c
---
 drivers/net/r8169.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index b2bee17..4cbc7be 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -6015,6 +6015,9 @@ static void rtl_set_rx_mode(struct net_device *dev)
 		mc_filter[1] = swab32(data);
 	}
 
+	if (tp->mac_version == RTL_GIGA_MAC_VER_35)
+		mc_filter[1] = mc_filter[0] = 0xffffffff;
+
 	RTL_W32(MAR0 + 4, mc_filter[1]);
 	RTL_W32(MAR0 + 0, mc_filter[0]);
 
-- 
1.8.1.4

